luigi/contrib/bigquery_avro.py

```diff
@@ -86,10 +86,64 @@ def get_fields_with_description(bq_fields, avro_fields):
-                        get_fields_with_description(field[u'fields'], avro_field.type.fields_dict)
+
+                #Enum Support
+                if field_type is avro.schema.EnumSchema:
+                    field[u'description'] = avro_field.type.doc
+
+                #Record Support
+                if field_type is avro.schema.RecordSchema:
+                    field[u'description'] = avro_field.type.doc
+                    field[u'fields'] = get_fields_with_description(field[u'fields'], avro_field.type.fields_dict)
+
+                #Array Support
+                if field_type is avro.schema.ArraySchema:
+                    field[u'description'] = avro_field.type.items.doc
+                    field[u'fields'] = get_fields_with_description(field[u'fields'], avro_field.type.items.fields_dict)
+
+                #Union Support
+                if type(avro_field.type) is avro.schema.UnionSchema:
+                    for schema in avro_field.type.schemas:
+                        if type(schema) is avro.schema.RecordSchema:
+                            field[u'description'] = schema.doc
+                            field[u'fields'] = get_fields_with_description(field[u'fields'], schema.fields_dict)
+
+                #Map Support
+                if field_type is avro.schema.MapSchema:
+                    #directly bypass the key and value attributes in BigQuery Schema
+                    bq_map_value_field = field[u'fields'][-1]
+                    avro_map_value = avro_field.type.values
+                    value_field_type = type(avro_map_value)
+
+                    #Primitive Support
+                    # -- Unfortunately the map element doesn't directly have a doc attribute so there is no way to get documentation on the primitive types for the value attribute
+
+                    if value_field_type is avro.schema.EnumSchema:
+                        bq_map_value_field[u'description'] = avro_map_value.type.doc
+
+                    if value_field_type is avro.schema.RecordSchema:
+                        #Set values description using type's doc
+                        bq_map_value_field[u'description'] = avro_map_value.doc
+
+                        #This is jumping into the map value directly and working with that
+                        bq_map_value_field[u'fields'] = get_fields_with_description(bq_map_value_field[u'fields'], avro_map_value.fields_dict)
+
+                    if value_field_type is avro.schema.ArraySchema:
+                        bq_map_value_field[u'description'] = avro_map_value.items.doc
+                        bq_map_value_field[u'fields'] = get_fields_with_description(bq_map_value_field[u'fields'], avro_map_value.items.fields_dict)
+
+                    #Unions nested in maps are not supported
+                    if value_field_type is avro.schema.UnionSchema:
+                        logger.info("Unions inside of maps are not supported for documentation.")
```

Comment:
Probably no need to make info logs noisy with this. Just explain they're not supported in the docstring of this task, https://github.com/spotify/luigi/pull/2224/files#diff-ce7b46ced67552a1fc5e2bdc4ccb11faL22

Reply:
added the details above and removed all excess logging